lib/cf-socket.c

```diff
@@ -1602,6 +1609,12 @@ CURLcode Curl_cf_tcp_create(struct Curl_cfilter **pcf,
     result = CURLE_OUT_OF_MEMORY;
     goto out;
   }
+
+#ifdef __linux__
```

Comment:
I think it is unwise to limit the MPTCP support for Linux only. AFAIK macOS/iOS already support it and FreeBSD is not far behind.

Reply:
It would be great to support other OS. But unfortunately, it is only easily accessible with Linux, thanks to `IPPROTO_MPTCP` that is for the moment specific to Linux

On iOS, MPTCP is available, but:
- It is easy only if you use their SDK: [doc](https://developer.apple.com/documentation/foundation/nsurlsessionconfiguration/improving_network_reliability_using_multipath_tcp)
- If not, you need to use private libraries (we are not even sure the headers are available) with specific functions to create sockets that are apparently not documented, e.g., [OpenSSH for iOS](https://github.com/apple-oss-distributions/OpenSSH/blob/main/openssh/sshconnect.c#L487)

On FreeBSD, there was an ongoing implementation, but that was years ago, and not working today according to [this](http://www-cs-students.stanford.edu/~sjac/freebsd_mptcp_info.html)

There are other implementations, but on specific systems (Citrix load balancer, userspace, etc.): more details [here](http://blog.multipath-tcp.org/blog/html/2018/12/15/apple_and_multipath_tcp.html)

So it looks like we might have to restrict this to Linux only for the moment. Is it an issue? It is always possible to add a `#elif` later, no?